documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,83 @@
 // Module included in the following assemblies:
 //
-// assembly-oauth.adoc
+// assembly-oauth-authentication.adoc
 
 [id='con-oauth-authentication-flow-{context}']
-= OAuth 2.0 authentication mechanism
+= OAuth 2.0 authentication mechanisms
 
-The Kafka _SASL OAUTHBEARER_ mechanism is used to establish authenticated sessions with a Kafka broker.
+Strimzi supports the OAUTHBEARER and PLAIN mechanisms for OAuth 2.0 authentication. 
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+The authentication flow between clients, the authorization server, and Kafka brokers is different for each mechanism.
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+Configure clients to use OAUTHBEARER for authentication with Kafka brokers whenever possible. 
+This mechanism provides a higher level of security because client credentials are _never_ shared with Kafka brokers. 
+Consider using PLAIN only with Kafka clients that do not support OAUTHBEARER. 
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+OAUTHBEARER and PLAIN can be enabled at the same time, on the same `oauth` listener, if both types of OAuth clients need to authenticate.
+
+.OAUTHBEARER overview
+
+Support for OAUTHBEARER is built in to many Kafka clients. 
+However, the developer must often write custom callbacks that provide access tokens for authentication, as needed.
+Strimzi provides these callbacks for standard Java Kafka clients (clients that use the Kafka producer and consumer APIs). 
+Therefore, developers do not need to provide custom callbacks in their application code. 
+
+With OAUTHBEARER, the client initiates a session with the Kafka broker for credentials exchange, where credentials take the form of a bearer token provided by the Strimzi OAuth callback code. 
+Using the callbacks, you can configure token provision in one of three ways:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+* A long-lived refresh token, obtained manually at configuration time
+
+OAUTHBEARER is automatically enabled in the `oauth` listener configuration for the Kafka broker. 
+You can set the `enableOauthBearer` property to `true`, though this is not required.
+
+[source,yaml,subs="attributes+"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    enableOauthBearer: true
+----
+
+.PLAIN overview
+
+PLAIN is a simple authentication mechanism supported by all Kafka client tools (including developer tools such as kafkacat). 
```

Comment:
We say earlier that _Support for OAUTHBEARER is built in to many clients_ and here we say PLAIN is supported by all _client tools_. Does this mean we would normally have to configure OAUTHBEARER, both or neither for clients and client tools?

Reply:
I don't know about the choice of workd but let me just clarify a bit ...
Many client tools use libraries that have basic OAUTHBEARER support (they understand it at protocol level), but require the client to write and install a callback handler that provides the access token. We provide such callback handler for upstream Kafka Client java library, but not for any other Kafka client library in other languages like Go, C/C++, JS ... In that case the client would have to write their own code for connecting to authorization server and obtaining the token.
Some Kafka client libraries may not even support OAUTHBEARER at protocol level or handle it properly by providing a callback mechanism. In that case the client simply can't use OAUTHBEARER.

Reply:
To reduce the ambiguity, I changed "built in" and "supported" to just "used". Apparently all clients use PLAIN. I added a new sentence: "OAUTHBEARER can only be used for Kafka broker authentication by Kafka clients that support this mechanism".